sdk-rs/src/store/adb.rs

```diff
@@ -0,0 +1,192 @@
+use crate::{Client as SdkClient, Error};
+use base64::{engine::general_purpose, Engine};
+use commonware_codec::FixedSize;
+use commonware_cryptography::{Hasher, Sha256};
+use commonware_storage::mmr::{hasher::Standard, verification::Proof};
+use http::HeaderMap;
+use reqwest::Response;
+use serde::{Deserialize, Serialize};
+use serde_with::{base64::Base64, serde_as};
+
+type Sha256Digest = <Sha256 as Hasher>::Digest;
+
+pub const PATH: &str = "/adb";
+
+pub struct Client {
+    client: SdkClient,
+    base_url: String,
+}
+
+/// The JSON response payload for a `get` adb operation. The payload provides both the value for the
+/// requested key, and a proof that should verify against the database's root corresponding to the
+/// provided MMR size.
+#[serde_as]
+#[derive(Serialize, Deserialize, Debug)]
+pub struct GetResultPayload {
+    /// The requested key's value.
+    #[serde_as(as = "Base64")]
+    pub value: Vec<u8>,
+
+    /// The position of the value in the MMR.
+    pub position: u64,
+
+    /// The raw proof data for the value that can be verified against the database's root at the
+    /// state corresponding to the provided MMR size.
+    #[serde_as(as = "Base64")]
+    pub proof_data: Vec<u8>,
+}
+
+impl Client {
+    pub fn new(client: SdkClient, parent_url: &str) -> Self {
+        Self {
+            client,
+            base_url: format!("{parent_url}{PATH}"),
+        }
+    }
+
+    pub async fn get_and_verify_proof(
+        &self,
+        root: [u8; 32],
```

Comment:
The rationale for not using `Digest` here is that we want to be agnostic to what is actually used?

Reply:
If so, I think we should remove the `<Sha256 as Hasher>::Digest` usage below.

For clarity, I'd actually be ok to hardcode `Sha256` for now (I haven't figured out how we should manage generic endpoints here yet).